### PR TITLE
7678-Remove-NuGet-References 

### PR DESCRIPTION
**Pull request by chris31389**
#### 📲 What

removing the NuGet package references and replacing with shared project references.
moving some projects to the src level where they are not coupled to the main project.

#### 🤔 Why

Its step 2 towards updating to .NET 8

#### 🛠 How

More in-depth discussion of the change or implementation.

#### 👀 Evidence

Screenshots / external resources / links / etc.
Link to documentation updated with changes impacted in the PR

#### 🕵️ How to test

Notes for QA

#### ✅ Acceptance criteria Checklist

- [ ] Code peer reviewed?
- [ ] Documentation has been updated to reflect the changes?
- [ ] Passing all automated tests, including a successful deployment?
- [ ] Passing any exploratory testing?
- [ ] Rebased/merged with latest changes from development and re-tested?
- [ ] Meeting the Coding Standards?

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -347,7 +360,6 @@ coverage.opencover.xml
 .store/
 dotnet-sonarscanner
 reportgenerator
```
Comment (sharonrussell) [inline .gitignore:362]: I don't think we should allow these files to be included, they could contain sensitive information

Comment (chris31389) [reply .gitignore:362]: for sensitive information i'd suggest we encorage the use of dotnet's user-secrets.

I needed to add some of these files to get the tests passing.  The Testing.Tests test the configuration and require files like this.

launchsettings.json is how the program is run, i've found it helpful in the past for all devs to share the same options, happy to be challenged on this.

If your keen to add these back in, we can do so now that the files we needed for the Testing.Tests are added

File: src/cqrs/src/tests/Functional/xxAMIDOxx.xxSTACKSxx.API.FunctionalTests/xxAMIDOxx.xxSTACKSxx.API.FunctionalTests.csproj [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
   <PropertyGroup>
     <TargetFramework>net6.0</TargetFramework>
+	  <IsTestProject>true</IsTestProject>
```
[the remark this answers is not in the document]
Comment (ElvenSpellmaker) [reply src/cqrs/src/tests/Functional/xxAMIDOxx.xxSTACKSxx.API.FunctionalTests/xxAMIDOxx.xxSTACKSxx.API.FunctionalTests.csproj:5]: This is something I've added on master, there's a bug in the Test SDK that doesn't inject that parameter properly when using `--no-restore` or `--no-build`, I'm on a couple of issue chains talking about it. I need to give someone from the SDK team more info to debug it further.

[the remark this answers is not in the document]
Comment (ElvenSpellmaker) [reply src/cqrs/src/tests/Functional/xxAMIDOxx.xxSTACKSxx.API.FunctionalTests/xxAMIDOxx.xxSTACKSxx.API.FunctionalTests.csproj:5]: Basically vstest won't run tests unless that parameter is present

**Review by sharonrussell — APPROVED** [review]

**Review by alexanderwjrussell — APPROVED** [review]
Looks good to me. 

Just the small change to the `.gitignore` I think would be good to change back.

File: src/cqrs/src/tests/Functional/xxAMIDOxx.xxSTACKSxx.API.FunctionalTests/xxAMIDOxx.xxSTACKSxx.API.FunctionalTests.csproj (hunk 2 above)
[the remark this answers is not in the document]
Comment (alexanderwjrussell) [reply src/cqrs/src/tests/Functional/xxAMIDOxx.xxSTACKSxx.API.FunctionalTests/xxAMIDOxx.xxSTACKSxx.API.FunctionalTests.csproj:5]: Thanks for explaining Jack. 

I did a little digging and realised it was an inert value so fine. Appreciate you taking the time to comment though :)

File: .gitignore (hunk 1 above)
Comment (alexanderwjrussell) [reply .gitignore:362]: I think my concern with this is that on other projects these have been ignored and therefore Devs leave keys in them. This is then a risk for it being exposed to GitHub. Especially as it is a change from the Microsoft standard.

Comment (chris31389) [reply .gitignore:365]: re-added

**Review by alexanderwjrussell — APPROVED** [review]

**Pull request merged**